### PR TITLE
#165032010 Implement comment on drafts feature

### PR DESCRIPTION
**Pull request by TheDrizzyWay**
## Description
This PR adds a functionality that enables users comment on article drafts and also reply to comments in a thread. This is done in order to ensure that comments made on drafts do not show up alongside comments made on published articles

## Type of change
- New feature

Screenshot 2019-04-09 at 6 21 13 AM

## Checklist:
- [x]  Write unit tests for this feature
- [x]  Create a table for saving draft comments.
- [x] Modify controllers for creating comments.

## PIVOTAL BOARD
[#165032010](https://www.pivotaltracker.com/story/show/165032010)

### CODE REVIEW

File: server/controllers/comment.controller.js [hunk 1]
```diff
@@ -22,9 +27,16 @@ class CommentController {
...
-    const newcomment = await comments.create({
-      userid: id,
-      articleid: slug,
+
+    const findArticle = await Article.findOne({
+      where: { slug },
+    });
+    if (!findArticle) return Response(res, 404, 'Article not found.');
+
+    const table = findArticle.dataValues.published ? comments : DraftComment;
```
Comment (davidokonji) [inline server/controllers/comment.controller.js]: Good work, like the way you went about with the check for drafts

File: server/models/comments.js [hunk 2]
```diff
@@ -39,17 +27,17 @@ module.exports = (sequelize, DataTypes) => {
   }, {});
   comments.associate = function(models) {
     comments.belongsTo(models.User, {
-      foreignKey: 'userid',
-      otherKey: 'articleid',
-      through: 'comments',
-      as: 'userComments',
+      foreignKey: 'userId',
+      otherKey: 'articleId',
+      as: 'usercomments',
```
Comment (davidokonji) [inline server/models/comments.js:32]: Thanks for changing this

File: server/controllers/comment.controller.js [hunk 3]
```diff
@@ -22,9 +27,16 @@ class CommentController {
     const { id } = req.user;
     const { slug } = req.params;
```
Comment (Chiazokam) [inline server/controllers/comment.controller.js:28]: Hi @TheDrizzyWay please a little explanation on the use of 'slug'

Comment (TheDrizzyWay) [reply server/controllers/comment.controller.js:28]: We are using the article slug based on the endpoint specs in the repository's readme file. It states that the article slug should be a part of the endpoint and be used to reference the article.

File: server/controllers/comment.controller.js [hunk 4]
```diff
@@ -59,9 +71,14 @@ class CommentController {
...
       comment
     } = req.body;
-    const newThreadComment = await comments.create({
-      userid: id,
-      articleid: slug,
+
+    const findArticle = await Article.findOne({
+      where: { slug },
+    });
+    const table = findArticle.dataValues.published ? comments : DraftComment;
```
Comment (samorahno) [inline server/controllers/comment.controller.js]: I appreciate the fact that you used a ternary operator here, neat job

**Review by samorahno — COMMENTED** [review]
Looks good, Great job

File: server/controllers/comment.controller.js [hunk 5]
```diff
@@ -59,9 +71,14 @@ class CommentController {
...
+    const newThreadComment = await table.create({
```
Comment (geekdanitek) [inline server/controllers/comment.controller.js:79]: Good work on descriptive variable naming.

File: server/controllers/comment.controller.js [hunk 6]
```diff
@@ -22,9 +27,16 @@ class CommentController {
     const { id } = req.user;
     const { slug } = req.params;
     const { comment } = req.body;
-    const newcomment = await comments.create({
-      userid: id,
-      articleid: slug,
+
+    const findArticle = await Article.findOne({
```
Comment (paradisekelechi) [inline server/controllers/comment.controller.js]: Use proper variable naming convention for `findArticle`.

File: server/controllers/comment.controller.js (hunk 1 above)
Comment (paradisekelechi) [inline server/controllers/comment.controller.js]: Why is there no consistency in the table naming? 
We have `comments` and `DraftComment` tables.

**Review by paradisekelechi — CHANGES_REQUESTED** [review]

File: server/controllers/comment.controller.js (hunk 1 above)
Comment (TheDrizzyWay) [reply server/controllers/comment.controller.js]: @paradisekelechi David worked on creating the comments table and I did not want to edit his code. Should i go ahead and fix it?

Comment (paradisekelechi) [reply server/controllers/comment.controller.js]: It's required that there be consistency. Check what the standard is and effect the required changes. @TheDrizzyWay

**Review by paradisekelechi — APPROVED** [review]

**Pull request merged**